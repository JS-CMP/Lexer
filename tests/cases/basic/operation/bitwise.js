var a = 5 & 3;
var b = 5 | 3;
var c = 5 ^ 3;
var d = ~5;
var e = 5 << 1;
var f = 5 >> 1;
"var g = 5 >>> 1; Not supported";
console.log(a, b, c, d, e, f);
