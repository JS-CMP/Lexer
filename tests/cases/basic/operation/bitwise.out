#include "types/JsAny.hpp"
#include "global/global.hpp"
#include "types/objects/Types.hpp"
#include "customOperators/CustomOperators.hpp"

int main() {
    JS::Any a, b, c, d, e, f, g;
    a = JS::Any(5) & JS::Any(3);
    b = JS::Any(5) | JS::Any(3);
    c = JS::Any(5) ^ JS::Any(3);
    d = ~JS::Any(5);
    e = JS::Any(5) << JS::Any(1);
    f = JS::Any(5) >> JS::Any(1);
    g = JS::Any(5) URightShift JS::Any(1);
    console[u"log"](a, b, c, d, e, f);
    return 0;
}
